packages/core/react/use-chat.ts

```diff
@@ -159,10 +159,38 @@ const getStreamedResponse = async (
 
   if (isComplexMode) {
     while (true) {
-      const { done, value } = await reader.read()
+      const { done, value: partialValue } = await reader.read()
       if (done) {
         break
       }
+
+      let value = partialValue
+
+      if (value[value.length - 1] !== 10) {
+        // if the last character is not a newline (ASCII code 10), we need to read more chunks
+        let buffer = new Uint8Array(value.length * 2);
+        buffer.set(value, 0); // copy the first chunk into the buffer
+        let offset = value.length;
+        while (value[value.length - 1] !== 10) {
```

Comment:
We can eliminate this inner loop and reuse the outer loop by hoisting the buffer variable outside the outer loop. If the last char isn't a newline, `continue`. Once found, process the buffer.

Reply:
I refactored this all together to use an array to collect the chunks. Still, we can not get rid of an inner loop altogether, since the outer loop is what iterates over complete parsable lines and the inner loop iterates over the chunks in a single line.